Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,903 @@
+
+    if ($totalDBForPostgreSQLFS -eq 0) {
+        Write-Host "No Azure Database for PostgreSQL flexible server(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+  
+    Write-Host "Found [$($totalDBForPostgreSQLFS)] Azure Database for PostgreSQL flexible server(s)." -ForegroundColor $([Constants]::MessageType.Update)
+                                                                          
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # list for storing Azure Database for PostgreSQL flexible server(s) for which server parameter require_secure_transport in not ON
+    $DBForPostgreSQLFSWithNonSecureTLSVersionEnabled = @()
+
+    Write-Host "Separating Azure Database for PostgreSQL flexible server(s) for which only secure TLS version is not set or secure transport communication is disabled..." -ForegroundColor $([Constants]::MessageType.Info)
+
+    $DBForPostgreSQLFlexibleServerDetails | ForEach-Object {
+        if (-not ( (CheckIfOnlySecureTLSVersionConfigured($_.TLSVersion)) -and (CheckIfSSLConfigured($_.SecureTransportStatus)))) {
+            $DBForPostgreSQLFSWithNonSecureTLSVersionEnabled += $_
+        }
+        else {
+            $logResource = @{}
+            $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+            $logResource.Add("ResourceName", ($_.ResourceName))	
+            $logResource.Add("Reason", "TLS version(s) and secure transport configured on this Resource.")    	
+            $logSkippedResources += $logResource	
+        }
+    }
+   
+    $totalDBForPostgreSQLFSWithNonSecureTLSVersionEnabled = ($DBForPostgreSQLFSWithNonSecureTLSVersionEnabled  | Measure-Object).Count
+
+    if ($totalDBForPostgreSQLFSWithNonSecureTLSVersionEnabled -eq 0) {
+        Write-Host "No Azure Database for PostgreSQL flexible server(s) found with non-secure TLS version enabled and secure transport communication is disabled.. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::DoubleDashLine)
+        
+        if ($AutoRemediation) {
+            $logFile = "LogFiles\" + $($TimeStamp) + "\log_" + $($SubscriptionId) + ".json"
+            $log = Get-content -Raw -path $logFile | ConvertFrom-Json
+            foreach ($logControl in $log.ControlList) {
+                if ($logControl.ControlId -eq $controlIds) {
+                    $logControl.RemediatedResources = $logRemediatedResources
+                    $logControl.SkippedResources = $logSkippedResources
+                }
+            }
+            $log | ConvertTo-json -depth 10  | Out-File $logFile
+        }
+        return
+    }
+
+    Write-Host "Found [$($totalDBForPostgreSQLFSWithNonSecureTLSVersionEnabled )] Azure Database for PostgreSQL flexible server(s) for which non secure TLS version is enabled ." -ForegroundColor $([Constants]::MessageType.Update)
```

Comment:
Is this just for non secure TLS version is enabled and not the other parameter?

Reply:
Fixed